ebos/eclgenericcpgridvanguard.cc

```diff
@@ -262,12 +267,13 @@ void EclGenericCpGridVanguard<ElementMapper,GridView,Scalar>::doCreateGrids_(Ecl
     {
         const bool has_numerical_aquifer = eclState.aquifer().hasNumericalAquifer();
         int mpiSize = 1;
-        MPI_Comm_size(MPI_COMM_WORLD, &mpiSize);
+        MPI_Comm_size(grid_->comm(), &mpiSize);
         // when there is numerical aquifers, new NNC are generated during grid processing
         // we need to pass the NNC from root process to other processes
         if (has_numerical_aquifer && mpiSize > 1) {
             auto nnc_input = eclState.getInputNNC();
-            EclMpiSerializer ser(Dune::MPIHelper::getCollectiveCommunication());
+            const auto& comm_nnc = Opm::Parallel::Communication();
```

Comment:
This uses the default (WORLD) communicator, I believe it is wrong and the grid_->comm() should be used. However I have not been able to actually provoke a failure in testing.

Reply:
It's a broadcast across the given communicator, so as long as all processes in the comm reach this code, you won't see a failure.

Reply:
Well, the idea is that you will use only a subset of the processes and then you will get a deadlock. Just use `grid_->comm()` as above